### ISSUE TITLE
training accuracy cannot match ssd  on caffe

### ISSUE THREAD

**Issue by terrychenism** [zhreshold/mxnet-ssd#6]
Training for 200 epoch for batchsize 32, test on voc2007, only got 68 mAP. 

Also it is very strange if batchsize > 32 when I train it on 2 or more GPUs. I have tested batchsize=64 using 2 gpus and 128 for 4 gpus, also  I have try different parameters, the mAP is worse than batchsize = 32. I know caffe implementation for VGGNet use batch 32, but it is not reasonable for large batchsize but bad performance. Is there any special normalization in the loss layer?

**Comment by zhreshold** [comment]
Please try different lr and wd, since I actually don't know if they are consistent in mxnet and caffe.  I also decreased the lr every 20 epoch, which seems to be not a good option.

For large batch size, the learning rate is not divided by batch size, if the batch is very large, then the gradients might get quite big. Also, it's not 100% true that larger batch will generate better performance.

The normalization is quite simple, by default(valid normalization), the gradient of softmax loss and smoothl1 loss will be divided by number of valid counts, i.e., (num_pos_anchors + num_neg_anchors) and (num_pos_anchors * 4), respectively.
Tuning grad_scale in loss layer is doable, you are very welcome to do it if you are interested.

**Comment by terrychenism** [comment]
Thanks, I will tune the parameters.

**Comment by zhreshold** [comment]
Please let me know if you have new discoveries.

**Comment by terrychenism** [comment]
Updates: lr 0.001,  lr-epoch 50,  end-epoch 70 .   smooth-l1 grad_scale 1.0   softmax 3.0  I got 69.7mAP.
I will continue to tune the params.

**Comment by zhreshold** [comment]
Just a tip, in caffe impl, the valid norm == number of positive anchors, in my impl, the valid norm == number of (positive + negative) anchors, thus theoratically my lr should be 4 times the caffe version lr, which is 0.001 with batch-size 32, i.e. 0.004
However, set lr to 0.004 usually ends up with l1 loss nan in the first few iterations. Based on previous experiences, I think the lr scale is different between caffe and mxnet. 
Don't know if it helps, anyway, thanks for you experiments.

**Comment by terrychenism** [comment]
Thanks for your reminds. I tested lr 0.002 vs 0.001 with different grad scale,  the final results almost same.

**Comment by hdjsjyl** [comment]
@terrychenism , Can you publish your parameters of final similar results? I am studying SSD. Thank you very much.
